dice/app/controllers/dice_controller.rb

```diff
@@ -19,6 +19,8 @@ def roll
     response = http.request(request)
     data = JSON.parse(response.body)
 
+    OpenTelemetry::Context.detach(token)
```

Comment:
Do we need to detach? If it copies the existing context contents and adds to this one... detaching seems unnecessary?

I would need to look into the code of how they have implemented the attach a bit deeper.

Reply:
Interesting - it does say this in the docs of the function
```

      # Associates a Context with the caller's current Fiber. Every call to
      # this operation should be paired with a corresponding call to detach.
      #
      # Returns a token to be used with the matching call to detach
      ```

They made no mention of it in the call but yeah I think you are right though they don't explain why 🙃

Reply:
I actually saw an error log when I was running it without the call to `detach`. 🤷